rhino/src/main/java/org/mozilla/javascript/Parser.java

```diff
@@ -4094,21 +4222,37 @@ boolean destructuringArray(
                 continue;
             }
             Node rightElem = new Node(Token.GETELEM, createName(tempName), createNumber(index));
+
+            if (defaultValue != null)
+                setupDefaultValues(tempName, parent, defaultValue, setOp, transformer);
+
             if (n.getType() == Token.NAME) {
+                /* [x] = [1] */
```

Comment:
I really like these comments, they make it nice and clear which cases each path is dealing with.